flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSingleRowJoin.scala

```diff
@@ -46,6 +47,7 @@ class DataSetSingleRowJoin(
     rowRelDataType: RelDataType,
     joinCondition: RexNode,
     joinRowType: RelDataType,
+    joinType: JoinRelType,
```

Comment:
That's a good point. I think we have not been very consistent in the past when adding such checks in the `DataSetRelNodes`. I think the constructor would be the right place though, because there is definitely something wrong with the optimization process if the optimizer wants to consider a plan with an invalid operator.